src/status_im/ui2/screens/chat/composer/input.cljs

```diff
@@ -192,49 +192,59 @@
-                                       last-text-change
-                                       mentionable-users)
-         :on-change
-         (partial on-change last-text-change timeout-id mentionable-users refs chat-id sending-image)
-         :on-text-input (partial on-text-input mentionable-users chat-id)}
+  (let [cooldown-enabled?   (rf/sub [:chats/current-chat-cooldown-enabled?])
+        mentionable-users   (rf/sub [:chats/mentionable-users])
+        timeout-id          (reagent/atom nil)
+        last-text-change    (reagent/atom nil)
+        mentions-enabled?   (get @mentions-enabled? chat-id)
+        props               {:style (text-input-style chat-id)
+                             :ref (:text-input-ref refs)
+                             :max-font-size-multiplier 1
+                             :accessibility-label :chat-message-input
+                             :text-align-vertical :center
+                             :multiline true
+                             :editable (not cooldown-enabled?)
+                             :blur-on-submit false
+                             :auto-focus false
+                             :max-length chat.constants/max-text-size
+                             :placeholder-text-color (colors/theme-colors colors/neutral-40
+                                                                          colors/white-opa-30)
+                             :placeholder (if cooldown-enabled?
+                                            (i18n/label :cooldown/text-input-disabled)
+                                            (i18n/label :t/type-a-message))
+                             :default-value (get @input-texts chat-id)
+                             :underline-color-android :transparent
+                             :auto-capitalize :sentences
+                             :auto-correct false
+                             :spell-check false
+                             :on-content-size-change on-content-size-change
+                             :on-selection-change (partial on-selection-change
+                                                           timeout-id
+                                                           last-text-change
+                                                           mentionable-users)
+                             :on-change
+                             (partial on-change
+                                      last-text-change
+                                      timeout-id
+                                      mentionable-users
+                                      refs
+                                      chat-id
+                                      sending-image)
+                             :on-text-input (partial on-text-input
+                                                     mentionable-users
+                                                     chat-id)}
         input-with-mentions (rf/sub [:chat/input-with-mentions])
-        children (fn []
```

Comment:
It was a function, it's not bad at all, but if we're going to pass it as a child (or a sequence of children), we need to evaluate it to effectively return the hiccup.

It was not being evaluated, it'd be enough if we just added `()` when used, or removed this `(fn[] )` .

Why is this a `fn`? for the current implementation I think it's not required so just removed it

Reply:
In this example it doesn't look right to wrap the hiccup in an anonymous function. What you did seems correct @ulisesmac.